sympy/matrices/eigen.py

```diff
@@ -375,27 +375,21 @@ def _is_diagonalizable(M, reals_only=False, **kwargs):
 #G&VL, Matrix Computations, Algo 5.4.2
 def _householder_vector(x):
     if not x.cols == 1:
-        raise ValueError("Input must be a column matrix")
+            raise ValueError("Input must be a column matrix")
     v = x.copy()
-    v_plus = x.copy()
-    v_minus = x.copy()
-    q = x[0, 0] / abs(x[0, 0])
-    norm_x = x.norm()
-    v_plus[0, 0] = x[0, 0] + q * norm_x
-    v_minus[0, 0] = x[0, 0] - q * norm_x
+    if x[0, 0] != 0:
+        q = x[0, 0] / abs(x[0, 0])
```

Comment:
In general you can't assume that `a != 0` means that `a` is nonzero e.g.:
```julia
In [49]: e = cos(1)**2 + sin(1)**2 - 1                                                                                                         

In [50]: e == 0                                                                                                                                
Out[50]: False

In [51]: simplify(e)                                                                                                                           
Out[51]: 0
```
Zero-testing is hard and is something that we should avoid needing to do if at all possible.

Reply:
Couldn't resolve, Justification is given below